CHANGELOG.md

```diff
@@ -1,3 +1,304 @@
+  config       = file("alertmanager-config.yaml")
+  node_selector = {
+    "kubernetes.io/hostname" = "worker3"
+  }
+}
+```
+
+```hcl
+# old
+prometheus_operator_node_selector = {
+  "kubernetes.io/hostname" = "worker3"
+}
+
+# new
+operator {
+  node_selector = {
+    "kubernetes.io/hostname" = "worker3"
+  }
+}
+```
+
+### Baremetal features: User data changes and reprovisioning of worker nodes
+
+Baremetal platform now supports user data changes and reprovisioning of worker nodes based on user data changes.
+
+From Lokomotive v0.9.0 onwards, additional files are created in the cluster assests directory.
+The filename being the MAC address of the machines and the contents being the domain name.
+
+The following paths are supported:
+
+* No changes to the worker nodes
+
+  In such a scenario, only thing that needs to be done is the above mentioned change in `k8s_domain_name`.
+  By default, user data changes are ignored.
+
+* User data changes but no PXE reprovisioning of worker nodes(reprovisioning happens via SSH):
+
+  In such a scenario, Lokomotive reboots the worker nodes and applies the user data changes. To bring about
+  such a change:
+
+  1. Make user data changes(if any).
+  2. Set `ignore_worker_changes = false`.
+
+* User data changes and reprovisioning of worker nodes:
+
+  In such a scenario, Lokomotive forces reinstallation of worker nodes via PXE and applies the user data changes.
+  This requires a meaningful `pxe_commands` value configured for automation. To bring about such a change:
+
+  1. Make user data changes (if any).
+  2. Remove the file with worker node MAC address from cluster assets directory.
+  3. Set `ignore_worker_changes = false` in cluster configuration.
+  4. Set `pxe_commands` to appropriate value.
+
+  **NOTE**: Reprovisioning will reinstallation the operating system. If you have any stateful workloads running, this
+  step would result is data loss.
+
+### Updating from v0.8.0
+
+#### Cluster update steps
+
+> **NOTE:** Updating multiple Lokomotive versions at a time is not supported. If your cluster is running a version older than `v0.8.0`, update to `v0.8.0` first and only then proceed with the update to `v0.9.0`.
+
+Execute the following steps in your cluster configuration directory:
+
+1. Download and install the lokoctl binary by following the
+  [v0.9.0 installation guide](https://github.com/kinvolk/lokomotive/blob/v0.9.0/docs/installer/lokoctl.md)
+  and verify the version using `lokoctl version`:
+
+  ```bash
+  v0.9.0
+  ```
+
+##### Update steps for Equinix Metal (formerly Packet)
+
+   a. Backup the Terraform state file:
+   ```bash
+   cd $assets_dir/terraform
+
+   # If using local backend
+   cp terraform.tfstate backup.tfstate
+
+   # If using AWS S3 backend
+   terraform state pull > backup.state
+   ```
+
+   b. Update Terraform provider from `packethost/packet` to `equinix/metal`:
+   ```bash
+   terraform state replace-provider packethost/packet equinix/metal
```

Comment:
Does this make changes to the state stored in S3?

Reply:
I want to be sure and back up state before starting anything.